tests/include/detray/test/common/assert.hpp

```diff
@@ -0,0 +1,17 @@
+/** Detray library, part of the ACTS project (R&D line)
+ *
+ * (c) 2024 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#include <gtest/gtest.h>
+
+#pragma once
+
+#define EXPECT_POINT3_NEAR(point1, point2, abs_error) \
+    do {                                              \
+        EXPECT_NEAR(point1[0], point2[0], abs_error); \
+        EXPECT_NEAR(point1[1], point2[1], abs_error); \
+        EXPECT_NEAR(point1[2], point2[2], abs_error); \
+    } while (false)
```

Comment:
What does this while loop do?

Reply:
It doesn't do anything, but this is a common macroing trick to ensure that a macro has to be followed by a semicolon. The loop runs only once, but it means that `EXPECT_POINT3_NEAR(a, b)` is invalid syntax while `EXPECT_POINT3_NEAR(a, b);` is valid.

Reply:
Ok